test/integration/kubernetes_deploy_test.rb

```diff
@@ -927,4 +927,23 @@ def test_resource_watcher_raises_after_timeout_seconds
       "If you expected it to take longer than 5 seconds for your deploy to roll out, increase --max-watch-seconds."
     ], in_order: true)
   end
+
+  def test_adds_namespace_labels_to_statsd_tags
+    desired_tags = %W(context:#{KubeclientHelper::MINIKUBE_CONTEXT} namespace:#{@namespace} statsd_tag:test)
+    hello_cloud = FixtureSetAssertions::HelloCloud.new(@namespace)
+    kubeclient.patch_namespace(hello_cloud.namespace, metadata: { labels: { statsd_tag: 'test' } })
+    metrics = capture_statsd_calls do
+      assert_deploy_success(deploy_fixtures("hello-cloud"))
+    end
+    metrics.each do |metric|
+      applicable = false
+      metric.tags.each do |metric_tag|
+        applicable = true if metric_tag.include?("test_adds_namespace_labels_to_statsd_tags")
```

Comment:
Since the way statsd-instrument captures its metrics is just by mocking a backend, we can't ensure that all the metrics we grab are from this specific test because they are running in parallel. The hacky solution I have is to make sure that the tag includes our test name.

Source: https://github.com/Shopify/statsd-instrument/blob/46df2c67ec9a9384b9b667df13497a884d560278/lib/statsd/instrument/helpers.rb#L3